mempool/v1/mempool_test.go

```diff
@@ -673,6 +672,40 @@ func TestMalleatedTxRemoval(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestBlobTxRemoval(t *testing.T) {
+	txmp := setup(t, 500)
+	originalTx := []byte{1, 2, 3, 4}
+	malleatedTx, err := types.WrapMalleatedTx(100, originalTx)
+	require.NoError(t, err)
+	b := tmproto.Blob{
+		NamespaceId:  []byte{1, 2, 3, 4, 5, 6, 7, 8},
+		Data:         []byte{1, 2, 3, 4, 5, 6, 7, 8, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		ShareVersion: 0,
+	}
+
+	// create the wrapped child transaction
+	bTx, err := types.WrapBlobTx(originalTx, &b)
+	require.NoError(t, err)
+
+	// add the parent transaction to the mempool
+	err = txmp.CheckTx(bTx, nil, mempool.TxInfo{})
+	require.NoError(t, err)
+	// make sure that we're not adding the blob transaction to the hash
+	require.False(t, txmp.cache.Has(bTx))
+
+	// remove the parent from the mempool using the malleated tx (tx w/ added metadata)
```

Comment:
```suggestion
	// remove the blob transaction from the mempool using the malleated tx (tx w/ added metadata)
```

[in a future PR] malleated transaction => block transaction